src/org/omegat/util/Preferences.java

```diff
@@ -762,8 +762,8 @@ public static synchronized void initSegmentation() {
         }
         didInitSegmentation = true;
 
-        File srxFile = new File(StaticUtils.getConfigDir(), SRX.CONF_SENTSEG);
-        SRX s = SRX.loadSRX(srxFile);
+        File srxDir = new File(StaticUtils.getConfigDir());
+        SRX s = SRX.loadSRX(srxDir); // may read SRX or CONF
```

Comment:
I personally would err on the side of *not* documenting the behavior of `loadSRX` outside of the method itself, but if you feel like it's surprising and needs to be mentioned then that's OK.

Reply:
I would say that the problem is the name of the method.
Previous version loaded only conf format, this was not in the documentation but since the previous line explicitly mentioned a CONF file, it could aspect clear enough.
Now that the previous line only mentions a dir, either we document it here, or we may search a better name for the method, what do you think?

Reply:
I do think the method could be renamed something like `loadSRXFromDir`. But I don't think it's relevant to the Preferences class whether the result came from a CONF or an SRX file.

Reply:
Better loadSegmentationFromDir, no? The idea is to avoid to use the name "SRX" for a method which _may_ load from another format.

Reply:
The class `SRX` predates any support for the actual SRX XML serialization format, so I think the best way to understand it is as a synonym for "segmentation".

Since `loadSRX` is a method on the class `SRX` and it returns an instance of `SRX`, I think having "SRX" in the method name is fine. The caller need not know or care about what the underlying serialization format was. However it's also a bit redundant; a good compromise would be to simply name it `loadFromDir` since the call will be `SRX.loadFromDir()` and thus obvious what we are loading.

I think the `SRX` class should probably be renamed to `Segmentation` or `SegmentationRules` or something like that.

Reply:
>  The class SRX predates any support for the actual SRX XML serialization format, so I think the best way to understand it is as a synonym for "segmentation".
> I think the SRX class should probably be renamed to Segmentation or SegmentationRules or something like that.

For me this is a bad idea, because SRX is not the only format/algorithm existing for segmentation.
Using this name while supporting only your old serializer (conf) format was not a problem because this format is semantically equivalent to SRX (i.e. you can convert from/to SRX without loosing any kind of information)
In another project I have created own segmentation formats, which give some possibilities better than SRX:
http://www.silvestris-lab.org/node/34
(description and samples are in the downloadable zip)
In the future I plan to implement them in DGT-OmegaT, simply because we have in DGT some rules which are absolutely impossible to express in SRX (for example "never cut inside a parenthesis"). I say in the future because the specification is not complete for now.
I did not begin this work but I am pretty sure that when I will do it, I will probably keep all other classes (MapRule, LanguageRule, Rule, and even Segmenter) unchanged, but simply replace the class SRX by a new one dedicated to my formats, and maybe add some classes for specific features.
So, for me SRX is the correct class name, because it means: in-memory representation of SRX file (also of a CONF file because the format is semantically equivalent) by opposition to any other segmentation format.

> a good compromise would be to simply name it loadFromDir

Yes, considering the fact that the method is static, it will always be preceeded by the class name, so no need to repeat it.
However, note the "from" in the name. Calling the method loadSRX, as actually, is correct in the sense that "you load SRX data". But if you have named it "loadFromSRX", then it would have been an error, because the method can load something else. 
Definitively I prefer loadFromDir. This is also coherent with the eventuality that, in the future, we load more than one file (for example, to have the list of abbreviations in a separate file)

Reply:
OK, renaming to loadFromDir is done.

Reply:
> I did not begin this work but I am pretty sure that when I will do it, I will probably keep all other classes (MapRule, LanguageRule, Rule, and even Segmenter) unchanged, but simply replace the class SRX by a new one dedicated to my formats, and maybe add some classes for specific features. Or, if I want to keep compatibility with all formats, I could have an abstract class SegmentationRules for which SRX is one implementation.

Finally I did it: implemented a (currently limited) support for CSCX format in DGT-OmegaT. And this works exactly as I had described before: an interface ISegmentationData which contains list of methods any format should support, then two implementations - SRX.java, which supports old CONF format because it is semantically equivalent; and CSC.java, which implements the specific part. Segmenter, LanguageRule and other classes are currently quite not affected.

I do not say that you should support such a format in OmegaT, that is not the question. But it confrms what I said about the name of SRX class.